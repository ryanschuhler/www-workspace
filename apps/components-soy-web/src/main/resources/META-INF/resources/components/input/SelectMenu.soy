{namespace SelectMenu}

/**
 * SelectMenu component
 */
{template .render}
  {@param name: string}
  {@param options: list<[
  label: string,
  value: any
  ]>}
  {@param? value: any}

  {let $none kind="text"}
    {msg desc=""}none{/msg}
  {/let}

  {let $selectMenuAttributes kind="attributes"}
    class="mdc-select"
    id="{$name}"
    name="{$name}"
    ref="selectMenu"
    role="listbox"
    tabindex="0"
    value="{$value}"
  {/let}

  <div {$selectMenuAttributes}>
    <span class="mdc-select__selected-text"></span>
    <div class="mdc-simple-menu mdc-select__menu">
      <ul class="mdc-list mdc-simple-menu__items">
        {foreach $option in $options }
          {let $attr kind="attributes"}
            {if $value == $option.value}
              aria-selected="true"
            {/if}

            class="mdc-list-item"
            id="{$option.value}"
            label="{$option.label}"
            onClick="handleOnClick_"
            role="option"
            tabindex="0"
            value="{$option.value}"
          {/let}

          <li {$attr}>
            <span class="padded-menu-span">
              {$option.label}
            </span>
          </li>
        {/foreach}
      </ul>
    </div>
  </div>
{/template}
