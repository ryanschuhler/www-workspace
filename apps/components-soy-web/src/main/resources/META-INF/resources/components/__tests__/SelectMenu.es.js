'use strict';

import SelectMenu from '../input/SelectMenu.es';

let component;

describe('SelectMenu', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {
			mdc: {
				select: {
					MDCSelect: jest.fn()
				}
			},
			once: jest.fn()
    };
  });

  it('should render', () => {
    component = new SelectMenu({
      name: 'foo',
      options: [{
        label: 'foo',
        value: 'bar'
      }]
    });

    expect(component).toMatchSnapshot();
  });
});