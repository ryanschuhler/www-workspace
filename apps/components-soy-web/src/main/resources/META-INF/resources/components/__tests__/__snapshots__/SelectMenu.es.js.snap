// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMenu should render 1`] = `
<div class="mdc-select" id="foo" name="foo" ref="selectMenu" role="listbox" tabindex="0">
  <span class="mdc-select__selected-text"></span>
  <div class="mdc-simple-menu mdc-select__menu">
    <ul class="mdc-list mdc-simple-menu__items">
      <li class="mdc-list-item" id="bar" label="foo" role="option" tabindex="0" value="bar">
        <span class="padded-menu-span">foo</span>
      </li>
    </ul>
  </div>
</div>
`;
